eth_tester/backends/pyethereum/v20/main.py

```diff
@@ -0,0 +1,208 @@
+
+class PyEthereum20Backend(BaseChainBackend):
+    tester_module = None
+
+    def __init__(self):
+        if not is_pyethereum20_available():
+            version = get_pyethereum_version()
+            if version is None:
+                raise pkg_resources.DistributionNotFound(
+                    "The `ethereum` package is not available.  The "
+                    "`PyEthereum20Backend` requires a 2.0.0+ version of the "
+                    "ethereum package to be installed."
+                )
+            elif version not in Spec('>=2.0.0,<2.2.0'):
+                raise pkg_resources.DistributionNotFound(
+                    "The `PyEthereum20Backend` requires a 2.0.0+ version of the "
+                    "`ethereum` package.  Found {0}".format(version)
+                )
+        from ethereum.tools import tester
+        self.tester_module = tester
+        self.evm = tester.Chain()
+        # NOTE: This is probably bad, but it solves get/set_fork_block() issues
+        self.evm.env = self.tester_module.get_env(None)
+    #
+    # Snapshot API
+    #
+    def take_snapshot(self):
+        return self.evm.snapshot()
+
+    def revert_to_snapshot(self, snapshot):
+        return self.evm.revert(snapshot)
+
+    def reset_to_genesis(self):
+        # NOTE: Not sure if this is right,
+        #       but it does reset to genesis
+        self.evm = tester.Chain()
+
+    #
+    # Fork block numbers
+    #
+    def set_fork_block(self, fork_name, fork_block):
+        if fork_name == FORK_HOMESTEAD:
+            self.evm.env.config['HOMESTEAD_FORK_BLKNUM'] = fork_block
+        elif fork_name == FORK_DAO:
+            self.evm.env.config['DAO_FORK_BLKNUM'] = fork_block
+        elif fork_name == FORK_ANTI_DOS:
+            self.evm.env.config['ANTI_DOS_FORK_BLKNUM'] = fork_block
+        elif fork_name == FORK_STATE_CLEANUP:
+            self.evm.env.config['CLEARING_FORK_BLKNUM'] = fork_block
+        else:
+            raise UnknownFork("Unknown fork name: {0}".format(fork_name))
+
+    def get_fork_block(self, fork_name):
+        if fork_name == FORK_HOMESTEAD:
+            return self.evm.env.config['HOMESTEAD_FORK_BLKNUM']
+        elif fork_name == FORK_DAO:
+            return self.evm.env.config['DAO_FORK_BLKNUM']
+        elif fork_name == FORK_ANTI_DOS:
+            return self.evm.env.config['ANTI_DOS_FORK_BLKNUM']
+        elif fork_name == FORK_STATE_CLEANUP:
+            return self.evm.env.config['CLEARING_FORK_BLKNUM']
+        else:
+            raise UnknownFork("Unknown fork name: {0}".format(fork_name))
+
+    #
+    # Meta
+    #
+    def time_travel(self, to_timestamp):
+        while to_timestamp >= self.get_state().timestamp:
+            self.mine_blocks()
+
+    #
+    # Mining
+    #
+    def mine_blocks(self, num_blocks=1, coinbase=None):
+        if not coinbase:
+            coinbase=self.get_accounts()[0]
+        block_hashes = []
+        for _ in range(num_blocks):
+            block_hashes.append(
```

Comment:
There is a utility in the `eth-utils` library, `eth_utils.to_tuple` which can be used as a decorator on this function to remove this build up of state by just using a yield statement here instead to return the mined blockhash.  Mutability, even benign mutability like this is something I prefer to avoid.

Reply:
Using now